xdsl/xdsl_opt_main.py

```diff
@@ -188,6 +189,7 @@ def register_all_dialects(self):
         self.ctx.register_dialect(LLVM)
         self.ctx.register_dialect(Vector)
         self.ctx.register_dialect(GPU)
+        self.ctx.register_dialect(Test)
```

Comment:
We might want to gate the testing dialect behind a special flag? I am unsure if we want it in the general `xdsl-opt` if that makes sense?

Reply:
I must say I am not sure at all and I had to think about this quite a lot to actually figure out an opinion. This is a very very weak opinion, but I don't feel like this is necessary. I don't think we lose anything from making it available per default and we add more complexity to the lit tests if we add a flag.

I also think there is value in tests being able to run from the command line without needing more than a few flags. Definitely a good and justified suggestions though! :)

Reply:
Yeah, I think you are right. This point will become moot anyway once our tests aren't the best source of documentation I guess :D